src/components/molecules/ChatsList/ChatsList.tsx

```diff
@@ -0,0 +1,179 @@
+import { WithId } from "utils/id";
+
+// Typings
+import { User } from "types/User";
+
+// Styles
+import "./ChatsList.scss";
+import { setPrivateChatMessageIsRead } from "components/organisms/PrivateChatModal/helpers";
+import { isEmpty } from "lodash";
+
+interface LastMessageByUser {
+  [userId: string]: PrivateChatMessage;
+}
+
+const ChatsList: React.FunctionComponent = () => {
+  const { user } = useUser();
+  const { privateChats, chatUsers } = useSelector((state) => ({
+    privateChats: state.firestore.ordered.privatechats,
+    chatUsers: state.firestore.data.chatUsers,
+  }));
+
+  const [selectedUser, setSelectedUser] = useState<WithId<User>>();
+
+  const discussionPartnerWithLastMessageExchanged =
+    privateChats &&
+    privateChats.reduce<LastMessageByUser>((agg, item) => {
+      let lastMessageTimeStamp;
+      let discussionPartner;
+      if (item.from === user?.uid) {
+        discussionPartner = item.to;
+      } else {
+        discussionPartner = item.from;
+      }
+      try {
+        lastMessageTimeStamp = agg[discussionPartner].ts_utc;
+        if (lastMessageTimeStamp < item.ts_utc) {
+          agg[discussionPartner] = item;
+        }
+      } catch {
+        agg[discussionPartner] = item;
+      }
+      return agg;
+    }, {});
+
+  const onClickOnSender = (sender: WithId<User>) => {
+    const chatsToUpdate = privateChats.filter(
+      (chat) => !chat.isRead && chat.from === sender.id
+    );
+    chatsToUpdate.map(
+      (chat) => user && setPrivateChatMessageIsRead(user.uid, chat.id)
+    );
+    setSelectedUser(sender);
+  };
+
+  const roundToNearestHour = (seconds: number) => {
+    const oneHour = 60 * 60;
+    return Math.floor(seconds / oneHour) * oneHour;
+  };
+  const DAYS_AGO = getDaysAgoInSeconds(VENUE_CHAT_AGE_DAYS);
+  const HIDE_BEFORE = roundToNearestHour(DAYS_AGO);
+
+  const chatsToDisplay = useMemo(
+    () =>
+      privateChats &&
+      privateChats
+        .filter(
+          (message) =>
+            message.deleted !== true &&
+            message.type === "private" &&
+            (message.to === selectedUser?.id ||
+              message.from === selectedUser?.id) &&
+            message.ts_utc.seconds > HIDE_BEFORE
+        )
+        .sort((a, b) => b.ts_utc.valueOf().localeCompare(a.ts_utc.valueOf())),
+    [privateChats, selectedUser, HIDE_BEFORE]
+  );
+
+  const chatContext = useContext(ChatContext);
+  const onMessageSubmit = async (data: { messageToTheBand: string }) => {
+    chatContext &&
+      user &&
+      chatContext.sendPrivateChat(
+        user.uid,
+        selectedUser!.id,
+        data.messageToTheBand
+      );
+  };
+
+  const hasPrivateChats = !isEmpty(discussionPartnerWithLastMessageExchanged);
+
+  return selectedUser ? (
+    <>
+      <div
+        className="private-container-back-btn"
+        onClick={() => setSelectedUser(undefined)}
+      />
+      <ChatBox chats={chatsToDisplay} onMessageSubmit={onMessageSubmit} />
+    </>
+  ) : (
+    <>
```

Comment:
In cases where the JSX has 'render one thing, or the other' at it's root, I tend to prefer this sort of style (see suggestion below), as I feel it keeps the logic cleaner/easier to see (and avoids having to nest the entire JSX tree).

Essentially, the first `return` in the component is what React will use. So if there is a single 'chunk' that we can return for a path, and then the other 'chunk' separately, this pattern works really well. If you also combine it with extracting sub-components from large chunks of JSX (eg. separating the frame/layout from the content, etc) then you can use this sort of pattern in more places. Which basically ends up with chunks of smaller/simpler to understand logic/etc.

```suggestion
  if (selectedUser) {
    return (
    <React.Fragment>
      <div
        className="private-container-back-btn"
        onClick={unsetSelectedUser}
      />
      <ChatBox chats={chatsToDisplay} onMessageSubmit={onMessageSubmit} />
    <React.Fragment/>
  }

  return (
    <React.Fragment>
    // ..snip..
    </React.Fragment>
```

Also, the callbacks will need `useCallback` as above.

(/cc @thecadams for interest RE: potential future code style things)

Reply:
I'm not too fussed about this; the alternative is less characters. It's personal preference. We should go with whatever an opinionated linter tells us; there may be a standard in eg. Airbnb's eslint/prettier rules, which we can rely on.

Reply:
+1 for automated linter assistance + automated fixes with prettier

Reply:
fixed in 70e8253242c4c56f3678a5035e0e6566912da044